### PR TITLE
Search update, ORM update, fixing the built modules

### PR DESCRIPTION
**Pull request by Sanne**
- https://issues.jboss.org/browse/ISPN-8779
 - https://issues.jboss.org/browse/ISPN-8838

Sending a more thourough explanation of this proposal to the mailing list.

### CODE REVIEW

File: wildfly-modules/pom.xml [hunk 1]
```diff
@@ -261,6 +222,24 @@
 
    <build>
       <plugins>
+         <plugin>
+            <groupId>org.wildfly.build</groupId>
+            <artifactId>wildfly-server-provisioning-maven-plugin</artifactId>
+            <version>1.2.6.Final</version>
```
Comment (ryanemerson) [inline wildfly-modules/pom.xml]: Nitpick, but could you uitlise the `${version.org.wildfly.build-tools}` property? It's currently on version `1.2.3.Final` but I can't imagine bumping this to `1.2.6.Final` causing any issues.

Comment (Sanne) [reply wildfly-modules/pom.xml]: thanks, I missed that this wasn't the first use.
I'll use the property as you suggest, and add a separate ticket to update it (which I include in this PR_

**Comment by Sanne** [comment]
> However, org.infinispan.query.blackbox.CompatModeClusteredCacheTest.testMerge seems related

That's an existing issue, I noticed the same but found it on ci.infinispan.org so I created a ticket for it yesterday: https://issues.jboss.org/browse/ISPN-8837?

**Comment by Sanne** [comment]
@ryanemerson updated the PR.

File: wildfly-modules/pom.xml [hunk 2]
```diff
@@ -188,49 +188,10 @@
...
-         <version>${version.hibernate.search}</version>
-         <classifier>wildfly-11-dist</classifier>
-         <type>zip</type>
-      </dependency>
-
       <dependency>
          <groupId>org.jboss.marshalling</groupId>
          <artifactId>jboss-marshalling-osgi</artifactId>
       </dependency>
-      <!-- Deps for the lucene module, will eventually be replaced by https://github.com/hibernate/lucene-modules -->
```
Comment (gustavonalle) [inline wildfly-modules/pom.xml]: I guess the time has finally come :)

**Comment by gustavonalle** [comment]
Running another CI, as it seems not all commits were tested....

**Comment by gustavonalle** [comment]
They actually were, my bad, integrating...

**Comment by gustavonalle** [comment]
thanks @Sanne !

**Pull request merged**